src/hotspot/cpu/ppc/c1_MacroAssembler_ppc.cpp

```diff
@@ -178,10 +178,7 @@ void C1_MacroAssembler::unlock_object(Register Rmark, Register Roop, Register Rb
   verify_oop(Roop, FILE_AND_LINE);
 
   if (LockingMode == LM_LIGHTWEIGHT) {
-    ld(Rmark, oopDesc::mark_offset_in_bytes(), Roop);
-    andi_(R0, Rmark, markWord::monitor_value);
-    bne(CCR0, slow_int);
-    lightweight_unlock(Roop, Rmark, slow_int);
+    lightweight_unlock(Roop, Rmark, Rbox, slow_int);
```

Comment:
I think using `Rbox` as temp register is not legal. `Runtime1::monitorexit` receives it as argument, so it must not be killed. I'll check if we can live with 1 temp reg less.